src/main/scala/io/iohk/ethereum/domain/BlockchainReader.scala

```diff
@@ -0,0 +1,105 @@
+package io.iohk.ethereum.domain
+
+import akka.util.ByteString
+import io.iohk.ethereum.db.storage.{
+  BlockBodiesStorage,
+  BlockHeadersStorage,
+  BlockNumberMappingStorage,
+  ReceiptStorage,
+  StateStorage
+}
+import io.iohk.ethereum.mpt.MptNode
+
+class BlockchainReader(
```

Comment:
would the next step be to create a trait `BlockchainReader` to allow for easier stubbing?

Reply:
Yes I agree. In the end we might split it further and have several reader traits.